src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -104,34 +104,32 @@ func onBootstrapWorkerNode(s *state.State, encodedToken string) error {
 		"--cluster-domain":    response.ClusterDomain,
 		"--cloud-provider":    response.CloudProvider,
 	}
-	if err := setup.InitKubeletArgs(snap, kubeletArgs, nil); err != nil {
+	if err := setup.InitKubeletArgs(sn, kubeletArgs, nil); err != nil {
 		return fmt.Errorf("failed to configure kubelet: %w", err)
 	}
-	if err := setup.RenderKubeletKubeconfig(snap, response.KubeletToken, response.CA); err != nil {
+	if err := setup.RenderKubeletKubeconfig(sn, response.KubeletToken, response.CA); err != nil {
 		return fmt.Errorf("failed to render kubelet kubeconfig: %w", err)
 	}
 
 	proxyArgs := map[string]string{
 		"--hostname-override": s.Name(),
 		"--cluster-cidr":      response.ClusterCIDR,
 	}
-	if err := setup.InitKubeProxyArgs(snap, proxyArgs, nil); err != nil {
+	if err := setup.InitKubeProxyArgs(sn, proxyArgs, nil); err != nil {
 		return fmt.Errorf("failed to configure kube-proxy: %w", err)
 	}
-	if err := setup.RenderKubeProxyKubeconfig(snap, response.KubeProxyToken, response.CA); err != nil {
+	if err := setup.RenderKubeProxyKubeconfig(sn, response.KubeProxyToken, response.CA); err != nil {
 		return fmt.Errorf("failed to render kube-proxy kubeconfig: %w", err)
 	}
 
-	if err := setup.InitAPIServerProxy(snap, response.APIServers); err != nil {
+	if err := setup.InitAPIServerProxy(sn, response.APIServers); err != nil {
 		return fmt.Errorf("failed to configure k8s-apiserver-proxy: %w", err)
 	}
 
 	// TODO: mark node as worker
 
-	for _, service := range []string{"containerd", "k8s-apiserver-proxy", "kubelet", "kube-proxy"} {
-		if err := snap.StartService(s.Context, fmt.Sprintf("k8s.%s", service)); err != nil {
-			return fmt.Errorf("failed to start service %s: %w", service, err)
-		}
+	if err := snap.StartServices(s.Context, sn, snap.WorkerServices); err != nil {
```

Comment:
better make this a `snap.StartWorkerServices`, and `snap.StartControlPlaneServices` below (make it hard for the caller to pass a wrong argument)

Reply:
also, I prefer `s` to `sn`, like we do in other places

Reply:
This was colliding with `s state` but just renamed `state` now.